### DISCUSSION TITLE
Brush save format

### DISCUSSION THREAD

**Discussion by MattBDev** [IntellectualSites/FastAsyncWorldEdit#943] [Ideas]
In previous versions of FAWE it was possible to load and save brushes that you used in game. We removed this because...well it was just implemented very poorly and was prone to breaking. Players found this feature particularly useful and were upset on it's removal.
The main concern about saving brushes was that there was no standard format and brushes could be saved and then not loaded. We need to create a spec that way this doesn't happen again in the future. 
Some things we need to address to re-add this functionality:

1. What type of file are we saving the brushes as? Are we using json, yaml, etc.?
2. Will the saved brush file have its own file extension, or will we stick to extension of the file format that we use?
3. What are the common properties of brushes currently in FAWE and WE?
4. What are some of the major differences of brushes in FAWE and WE?
5. What will brush files be named? Can a user name the file when they save the brush? Where will they be saved?

I'll update this issue with more questions as they come up so we can eventually start a PR with a spec like how schematics have a spec.
I am requesting feedback from all the @IntellectualSites/fawe-team on this because we need to get it right before an implementation is is compiled and available. 

**Users can help us with this by telling us what they expect to happen when they load and save brushes.**

**Comment by SirYwell** [comment]
Might be worth to keep an eye on https://github.com/EngineHub/WorldEdit/issues/1306 and probably discuss such specification with the WE team.

**Comment by MattBDev** [comment]
Well I'm just trying to get things started so we can start having the chats with them. As soon as some of these first questions are addressed, we can get to work on it and discuss details of what a saved tool/brush looks like.

**Comment by anonyco** [comment]
Hi,

Two proposals in my document https://github.com/IntellectualSites/FastAsyncWorldEdit/discussions/1886 come to mind for this:
1. Persistent brush saving
2. `//shortcut` command for quick access to running commands. You can make a shortcut that applies a brush to a tool
